pkg/kubectl/cmd/create/create.go

```diff
@@ -161,10 +160,8 @@ func (o *CreateOptions) ValidateArgs(cmd *cobra.Command, args []string) error {
 }
 
 func (o *CreateOptions) Complete(cmd *cobra.Command) error {
-	o.DryRun = cmdutil.GetDryRunFlag(cmd)
-
-	if o.DryRun {
-		o.PrintFlags.Complete("%s (dry run)")
+	if o.PrintFlags.DryRun != nil {
+		o.DryRun = *o.PrintFlags.DryRun
```

Comment:
A printer doesn't need to be aware of dry-running. It is told: "print this thing".  How the thing was produced and what actions were taken with it are not logically relevant to a printer.